src/libcore/yk_swt_impl.c

```diff
@@ -29,52 +29,52 @@ static __thread size_t trace_buf_len = 0;
 // The allocation capacity of the trace buffer (in elements).
 static __thread size_t trace_buf_cap = 0;
 // Is the current thread tracing?
-static __thread bool tracing = false;
-// This flag is set when the contents of the trace buffer may not be valid.
-static __thread volatile atomic_bool trace_invalid = false;
+// true = we are tracing, false = we are not tracing or an error occurred.
+static __thread volatile atomic_bool tracing = false;
 
 // Start tracing on the current thread.
-// A new trace buffer is allocated and MIR locations are written into it. The
-// current thread must not already be tracing.
+// A new trace buffer is allocated and MIR locations will be written into it on
+// subsequent calls to `yk_swt_rec_loc_impl`. The current thread must not
```

Comment:
Maybe change the last sentence to "If the current thread is already tracing, then calling this function will lead to undefined behaviour."?

Reply:
Sure. It will at least leak memory, but I guess there's no need to mention that. All under the umbrella of undefined behaviour :)

Reply:
Indeed!